packages/terra-table/src/terra-dev-site/doc/table/guides.91/CheckMarkTable.94.doc.mdx

```diff
@@ -0,0 +1,183 @@
+import CheckMarkTable from 'terra-table/lib/terra-dev-site/doc/guides/CheckMarkTable?dev-site-example';
+
+# Terra Table - Implementing a CheckMark Table
+
+The terra-table implementation requires controlled state if selections are required. As a result selections are applied at child row generation from HOC state. The following guide show you how to implement that state within a multiple row selection variant of table, as well as implement the additional narrowing requirements of a max selection for rows.
+
+## State Management
+The state of selection needs to be managed for the table in a High Order Component (HOC). In this example we are going to be using a unique key to manage the selection state, but the type of state used is open to the implementor of the HOC. The `selectedKeys` will keep track of the keys that are active given the allSelected state. In the case where `allSelected` is "false" the key values will be the keys that are selected, and in the case of "true" it will be the keys that are not selected. This dual behavior removes the need to copy all keys into state when the "select all" in checked.
```

Comment:
I don't think Higher-Order Component is the right term to use here. HOCs are specifically defined as components like those provided by `connect` or `injectIntl` that are isolated to a specific purpose for reusability purposes.

Reply:
Updated.